referenceframe/frame_system.go

```diff
@@ -308,6 +309,55 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 	return newFS, nil
 }
 
+// FrameSystemToPCD in a framesystem and returns a map where all elements are the point representation
+// of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input) (map[string][]r3.Vector, error) {
+	vectorMap := make(map[string][]r3.Vector)
+	for _, name := range system.FrameNames() {
+		currentFrame := system.Frame(name)
+		parent, err := system.Parent(currentFrame)
+		if err != nil {
+			return nil, err
+		}
+
+		geosInFrame, err := currentFrame.Geometries(inputs[name])
+		if err != nil {
+			return nil, err
+		}
+		if parent.Name() != "world" {
+			transformed, err := system.Transform(inputs, geosInFrame, World)
+			if err != nil {
+				return nil, err
+			}
+			transformedGeo := transformed.(*GeometriesInFrame)
+			var aggregatePoints []r3.Vector
+			for _, g := range transformedGeo.Geometries() {
+				aggregatePoints = append(aggregatePoints, g.ToPoints(1.)...)
+			}
+
+			parentFrame := system.Frame(parent.Name())
```

Comment:
I'm not clear on what this code with the Parent (L338-L347) is doing/why it's needed. Could you elaborate?

Reply:
the geometries within 'transformedGeo' have been transformed to be with respect to the world. however since their parent is not world they need to be translated to be located at their parent. I construct parentFrame so i can get the needed pose and then translate the  geometries of transformedGeo accordingly.

Reply:
Yeah the Transform call should be working the way Peter is suggesting (and if its not thats a problem), so most of this can probably go away